diesel/src/query_builder/sql_query.rs

```diff
@@ -164,3 +178,85 @@ where
 }
 
 impl<Conn, Query, Value, ST> RunQueryDsl<Conn> for UncheckedBind<Query, Value, ST> {}
+
+#[must_use = "Queries are only executed when calling `load`, `get_result`, or similar."]
+/// See `SqlQuery`
+pub struct BoxedSqlQuery<'f, DB: Backend, Inner> {
+    inner: Inner,
+    sql: String,
+    binds: Vec<Box<dyn Fn(AstPass<DB>) -> QueryResult<()> + 'f>>,
+}
+
+impl<DB: Backend, Inner> Debug for BoxedSqlQuery<'_, DB, Inner>
```

Comment:
cc @diesel-rs/core This seems to be a clever idea. Maybe we could do similar things for other query dsl types?

Reply:
It's an interesting idea, but I think it needs work to be added (delimiters, indicating what it actually is, etc). There's a discussion to be had here, but either way that should be a separate PR, not part of this one.

Reply:
Removed